### PR TITLE
Fixes #11119: Added check for wacky pet before feeding

### LINKED ISSUE #11119: Wacky Pets should not be feedable
[//]: # (Before logging this issue, please post to the Report a Bug guild from the Habitica website's Help menu. Most bugs can be handled quickly there. If a GitHub issue is needed, you will be advised of that by a moderator or staff member -- a player with a dark blue or purple name. It is recommended that you don't create a new issue unless advised to.)

[//]: # (Bugs in the mobile apps can also be reported there.)

[//]: # (If you have a feature request, use "Help > Request a Feature", not GitHub or the Report a Bug guild.)

[//]: # (For more guidelines see https://github.com/HabitRPG/habitica/issues/2760)

[//]: # (Fill out relevant information - UUID is found from the Habitia website at User Icon > Settings > API)

### Description
[//]: # (Describe bug in detail here. Include screenshots if helpful.)

Currently, the API doesn't reject attempts to feed Veggie pets; it goes ahead and increases their food counter and can even transform them into mounts, causing them to vanish because Veggie Mounts don't (yet?) exist. We head this interaction off on the client side for Web, but other API consumers might still attempt this. The API should error out and not remove food, etc. in these cases.

### PR DESCRIPTION
**Pull request by somersjw**
[//]: # (Note: See http://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #11119

### Changes
[//]: Added a check when on the server API in feed.js to display messageCannotFeedPet if the pet type is a wacky pet.

[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

----
UUID: 321dffb4-8124-40e4-832b-d32b5fb997fa

### CODE REVIEW

**Comment by paglias** [comment]
Thanks @somersjw ! The code looks good, could you add a test covering this new code? there's an existing ones for special pets that can be reused here https://github.com/HabitRPG/habitica/blob/develop/test/common/ops/feed.js#L80-L90 but let me know if you have any question!

**Comment by somersjw** [comment]
I added the test that you aksed for @paglias, but the test seems to be failing. Is there something I did wrong here?

File: test/common/ops/feed.js [hunk 1]
```diff
@@ -89,6 +89,18 @@ describe('shared.ops.feed', () => {
       }
     });
 
+    it('does not allow feeding of wacky pets', done => {
+      user.items.pets['Wolf-Veggie'] = 5;
+      user.items.food.Meat = 1;
+      try {
+        feed(user, { params: { pet: 'Wolf-Vegie', food: 'Meat' } });
```
Comment (paglias) [inline test/common/ops/feed.js]: looks like a `g` got lost here :) `Wolf-Vegie` vs `Wolf-Veggie`

Comment (somersjw) [reply test/common/ops/feed.js]: Well that was embarrassing! Thanks for catching that :)

Comment (paglias) [reply test/common/ops/feed.js]: no worries :)

**Comment by paglias** [comment]
@somersjw  thanks for this! I'm merging the PR and awarding your contributor tier.

Note that further tiers require increasing amounts of work from one to the next, so Tier 2 may require a couple of PRs or a larger PR to attain. But keep helping out and we'll express our gratitude accordingly!

**Pull request merged**